tests/testparameterizedobject.py

```diff
@@ -297,13 +344,109 @@ def test_missing_key(self):
+        assert overrides.get('print_level') == 0
+        
+        assert overrides.get('bogus') is None
+        assert overrides.get('bogus',20) is 20
+
+        assert [('name', overrides.name), ('print_level', 0), ('some_param', 'y')] == sorted(overrides.items())
+
+    def test_is_mapping(self):
+        def f(**kw): pass
+        overrides = ParamOverrides(self.po,{'some_param':'y'})
+        f(**overrides)
+    
+    # this test is about checking things are looked up normally (like
+    # for dynamic tests above) so is probably redundant now
+    # paramoverrides is itself a parameterized object.
+    def test_search_paths(self):
+        # TODO: only good on linux/max - should do this properly
+        test_file = 'passwd'
+        test_path = '/etc/'
+
+        ###
+        # this will fail if the test itsel isn't working
+        class A(param.Parameterized):
+            path = param.Path(search_paths=['/etc'])
+        a = A(path=test_file)
+        assert a.path==test_path+test_file
+        ###
+
+        # the actual test...
+        class A2(param.Parameterized):
+            path = param.Path(search_paths=['/etc'])
+            def __call__(self,**params):
+                p = param.ParamOverrides(self,params)
+                return p.path
+    
+        a2 = A2(path=test_file)
+        assert a2() == test_path+test_file, a2()
+        assert a2(path=test_file) == test_path+test_file, a2(path=test_file)
+
+
+    # another test to simplify one day!
+    def test_extra_keywords(self):
+        try:
+            old = param.parameterized.warnings_as_exceptions
+            param.parameterized.warnings_as_exceptions = True
+            p = param.ParamOverrides(self.po,{'extra':100},allow_extra_keywords=False)
+        except:
+            pass
+        else:
+            raise AssertionError
+        finally:
+            param.parameterized.warnings_as_exceptions = old
+            
+        p = param.ParamOverrides(self.po,{'extra':100},allow_extra_keywords=False)
+        assert len(p._extra_keywords) == 0
+
+        p = param.ParamOverrides(self.po,{'extra':100},allow_extra_keywords=True)
+        assert len(p._extra_keywords) == 1
+        assert p._extra_keywords['extra'] == 100
+
+    def test_nonparamattr(self):
+        x = AnotherTestPO(name='A')
+        x.some_other_thing = 1
+        assert x.some_other_thing == 1
+        overrides = ParamOverrides(x,{'some_param':'y'})
+        assert overrides.some_other_thing == 1
+        
+    def test_param_keywords(self):
+        overrides = ParamOverrides(self.po,{'name':'B','an_extra_thing':1},allow_extra_keywords=True)
+        assert 'an_extra_thing' in overrides.extra_keywords()
+        assert 'an_extra_thing' not in overrides.param_keywords()
+
+    def test_pickle(self):
+        overrides = ParamOverrides(self.po,{'name':'B','an_extra_thing':1},allow_extra_keywords=True)
+        import pickle
+        pkl = pickle.dumps(overrides)
```

Comment:
Should test pickle load also. (Note: pickle load seems to be broken for the current ParamOverrides anyway. Should investigate.)